### PR TITLE
versioning: transition from nerdbank to version file

### PR DESCRIPTION
**Pull request by ldennington**
With the [plan to migrate GCM to a formula for release via `Homebrew/homebrew-core`](https://github.com/git-ecosystem/git-credential-manager/issues/1102), it became clear that [Nerdbank.GitVersioning](https://github.com/dotnet/Nerdbank.GitVersioning) was no longer an option for versioning the project. This is because `Nerdbank.GitVersioning` requires history to calculate things like ['Git height'](https://github.com/dotnet/Nerdbank.GitVersioning#what-is-git-height), and the formula requires a tarball to build, which, of course, has no history.

This change pivots GCM to the simpler strategy of using a version specified in a `VERSION` file at root. This version will be updated by maintainers prior to release - giving them more granular control of versioning, which in turn allows for versioning to be more predictable (i.e. maintainers will know what the version will be before publication of the release).

The version specified in the file is the one slated for the next release: `2.1.0.0`.

**Note:** This change [fails on Windows](https://github.com/ldennington/git-credential-manager/actions/runs/4824686907) due to a [bug in MSAL](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4108) unless we ensure all projects have at least 1 dependency. We are working around this issue by adding Newtonsoft.Json as this dependency (since we were already shipping it anyway).

### CODE REVIEW

**Comment by ldennington** [comment]
Note: There are a lot of whitespace corrections that were automatically added to my change due to my VS Code settings.

**Review by mjcheetham — APPROVED** [review]
:shipit:

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -128,25 +125,25 @@ jobs:
...
+    - name: Set version environment variable
+      run: echo "VERSION=$(cat VERSION)" >> $GITHUB_ENV
 
     - name: Set up dotnet
       uses: actions/setup-dotnet@v3.0.3
       with:
         dotnet-version: 6.0.201
-    
+
     # Install Nerdbank.GitVersioning
```
[the remark this answers is not in the document]
Comment (ldennington) [reply .github/workflows/release.yml:147]: Great catch!

[the remark this answers is not in the document]
Comment (ldennington) [reply .github/workflows/release.yml]: Just pushed a new commit with these changes - planning to wait for my [test run](https://github.com/ldennington/git-credential-manager/actions/runs/4833207002) to complete before merging.

**Review by mjcheetham — APPROVED** [review]

**Review by mjcheetham — APPROVED** [review]

**Pull request merged**